frontend/src/components/elements/Favicon/Favicon.tsx

```diff
@@ -0,0 +1,99 @@
+/**
+ * @license
+ * Copyright 2018-2020 Streamlit Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React, { Component, ReactNode } from "react"
+import ReactDOM from "react-dom"
+import { Map as ImmutableMap } from "immutable"
+import { getImageURI } from "../ImageList/ImageList"
+import nodeEmoji from "node-emoji"
+
+export interface Props {
+  element: ImmutableMap<string, any>
+}
+
+/**
+ * Hidden element that overwrites the page's favicon with the provided image
+ */
+export class Favicon extends Component<Props> {
+  emoji = ""
+  shouldRender = true
+
+  public render(): ReactNode {
+    const { element } = this.props
+
+    const maybeEmoji = element.get("url")
+    if (nodeEmoji.hasEmoji(nodeEmoji.get(maybeEmoji))) {
+      // Format: :pizza:
+      this.emoji = nodeEmoji.get(maybeEmoji)
+    } else if (nodeEmoji.hasEmoji(maybeEmoji)) {
+      // Format: 🍕
+      this.emoji = maybeEmoji
+    } else {
+      // Format: http://streamlit.io/favicon.ico or /media/blah.jpeg
+      // No need to render SVG
+      this.setFavicon(getImageURI(element))
+      this.shouldRender = false
+    }
+
+    return this.shouldRender ? (
+      <svg
+        version="1.2"
+        viewBox="0 0 100 100"
+        xmlns="http://www.w3.org/2000/svg"
+      >
+        <text>{this.emoji}</text>
+      </svg>
+    ) : null
+  }
+
+  setFavicon(imageUrl: string) {
```

Comment:
same here

Reply:
Since this wasn't really a class method to begin with (didn't use any references of `this`), I've extracted it as a pure function.